quickwit/quickwit-ui/src/utils/models.ts

```diff
@@ -86,6 +86,100 @@ export type SearchRequest = {
   endTimestamp: number | null;
   maxHits: number;
   sortByField: SortByField | null;
+  aggregation: boolean;
+  aggregationConfig: Aggregation;
+}
+
+export type Aggregation = {
+  metric: Metric | null;
+  term: TermAgg | null;
+  histogram: HistogramAgg | null;
+}
+
+export type Metric = {
+  type: string;
+  field: string;
+}
+
+export type TermAgg = {
+  field: string;
+  size: number;
+}
+
+export type HistogramAgg = {
+  interval: string;
+}
+
+export type ParsedAggregationResult = TermResult | HistogramResult| null;
+
+export type TermResult = {term: string, value: number}[];
+
+export type HistogramResult = {
+  timestamps: Date[],
+  data: {name: string | undefined, value: number[]}[],
+}
+
+export function extractAggregationResults(aggregation: any): ParsedAggregationResult {
+  console.log("doing the conversion");
+  const extract_value = (entry: any) => {
+    if (Object.prototype.hasOwnProperty.call(entry, "metric")) {
```

Comment:
nit : `"metric" in entry` is less verbose.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in